### PR TITLE
Add greedy matcher to smart classroom demo

### PR DESCRIPTION
**Pull request by LeonidBeynenson**
The greedy matcher allows to run the demo in the cases when 
the number of gallery images is greater than 1000, although it
may decrease quality of face reidentification a bit.

### CODE REVIEW

**Comment by LeonidBeynenson** [comment]
2 @Ilya-Krylov : Ilya, could you review this PR?

**Comment by Ilya-Krylov** [comment]
:+1:

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

File: demos/smart_classroom_demo/include/smart_classroom_demo.hpp [hunk 1]
```diff
@@ -307,6 +313,7 @@ static void showUsage() {
     std::cout << "    -d_lm '<device>'               " << target_device_message_landmarks_regression << std::endl;
     std::cout << "    -d_reid '<device>'             " << target_device_message_face_reid << std::endl;
     std::cout << "    -out_v  '<path>'               " << output_video_message << std::endl;
+    std::cout << "    -greedy_reid_matching          " << greedy_reid_matching_message << std::endl;
```
[the remark this answers is not in the document]
Comment (LeonidBeynenson) [reply demos/smart_classroom_demo/include/smart_classroom_demo.hpp]: Fixed!

**Comment by LeonidBeynenson** [comment]
2 @vladimir-dudnik, @IRDonch: I've fixed README. If everything is OK, could you merge this PR?

**Comment by IRDonch** [comment]
Jenkins: test this please

**Pull request merged**